src/librustc/query/mod.rs

```diff
@@ -75,6 +75,7 @@ rustc_queries! {
         // This can be conveniently accessed by methods on `tcx.hir()`.
         // Avoid calling this query directly.
         query hir_owner(key: DefId) -> &'tcx HirOwner<'tcx> {
```

Comment:
Change the query keys to `LocalDefId`, then you won't have any `is_local` checks, maybe that helps?

Reply:
They are `LocalDefId` in `master` now. I don't expect removing a single branch to help much though.